polaris-react/.storybook/preview.js

```diff
@@ -28,7 +28,7 @@ function AppProviderDecorator(Story, context) {
 export const globalTypes = {
   strictMode: {
     name: 'React.StrictMode',
-    defaultValue: 'false',
+    defaultValue: 'true',
```

Comment:
I think we want to match `shopify/web`, if they use strict by default this would be fine to merge in.

Reply:
StrictMode has zero effect in production, it's just a development tool used to [check for correctness](https://reactjs.org/docs/strict-mode.html). I'd argue that helping ensure correct behaviour when developing in polaris-react is worthwhile even if web does not do that currently in its local development. Having polaris be strict mode compliant would make it easier for web and other consuming apps to adopt strict mode in their own codebases.

Reply:
It doesn't look like they do which seems weird. We've always defaulted to strict mode in Polaris though. @BPScott any thoughts on this? Maybe we enable strict mode for storybook web as well?